extensions/wikia/WikiFactory/SpecialWikiFactory_body.php

```diff
@@ -61,160 +65,201 @@ public function execute( $subpage ) {
+	 *   - The "wpCityDomain" query parameter if it has the format "id:1234"
+	 *
+	 * @param string $subpage The parts of the page after Special:WikiFactory
+	 *
+	 * @return String
+	 */
+	private function findCityId( $subpage ) {
+		// Look in the cityId query parameter first
+		$cityId	= F::app()->wg->Request->getVal( "cityId", null );
+		if ( !empty( $cityId ) ) {
+			return $cityId;
 		}
-		if ( !isset($this->mVariableName) ) {
-			$this->mVariableName = "";
+
+		$pathParts = explode( '/', $subpage );
+		if ( !empty( $pathParts[0] ) && ctype_digit( $pathParts[0] ) ) {
+			return $pathParts[0];
 		}
-		return WikiFactory::getWikiByID( $cityid );
+
+		return $this->findCityIdFromDomain( $subpage );
 	}
 
 	/**
-	 * doWikiSelector
+	 * Determine the cityId by looking for a domain.  The domain is found in the
+	 * wpCityDomain parameter and can have any of the following formats
 	 *
-	 * Pager with all wikis
+	 * - id:1234 : A city ID masquerading as a domain
+	 * - db:Callofduty : A dbName
+	 * - callofduty : For single names (no dot separators) add .wikia.com and try as domain
+	 * - callofduty.wikia.com : A full domain
+	 * - either of the previous two used in the subdomain
+	 * - http://callofduty.wikia.com/wiki/Page : Any pasted URL
+	 *
+	 * @param string $subpage
 	 *
-	 * @access private
-	 * @author eloy
+	 * @return mixed
+	 *
+	 * @throws Exception
+	 */
+	private function findCityIdFromDomain( $subpage ) {
+		// Get rid of any "http://" or path parts
+		$domain = $this->getCleanedDomain();
+
+		// Try to use the first path part as a domain
+		if ( empty( $domain ) ) {
+			$pathParts = explode( '/', $subpage );
+			if ( empty( $pathParts[0] ) ) {
+				return null;
+			}
+			$domain = $pathParts[0];
+		}
+
+		// Anything starting with id: is a city ID
+		if ( preg_match( "/^id:(\\d+)/", $domain, $matches ) ) {
+			return $matches[1];
+		}
+
+		// A value keyed with db: should be treated as a dbName
+		if ( preg_match( "/^db:(\\S+)/", $domain, $matches ) ) {
+			$dbName = $matches[1];
+			return WikiFactory::DBtoID( strtolower( $dbName ) );
+		}
+
+		// A single value, try it as the first part of a full domain
+		if ( preg_match( "/^[\\.]+$/", $domain ) ) {
+			$domain = $domain . "." . F::app()->wg->WikiaBaseDomain;
+			return WikiFactory::DomainToID( strtolower( $domain ) );
+		}
+
+		// Otherwise treat this like a full domain
+		$domain = WikiFactory::getLocalEnvServer( $domain, WIKIA_ENV_PROD );
+		return WikiFactory::DomainToID( strtolower( $domain ) );
+	}
+
+	private function getCleanedDomain() {
+		$domain	= F::app()->wg->Request->getVal( "wpCityDomain", null );
+
+		return preg_replace( '#^\s*https?://|/.*$|\s+$#', '', $domain );
+	}
+
+	/**
+	 * doWikiSelector
 	 *
-	 * @return nothing
+	 * Pager with all wikis
 	 */
 	private function doWikiSelector() {
 		global $wgOut;
 
 		$oTmpl = new EasyTemplate( dirname( __FILE__ ) . "/templates/" );
-		$oTmpl->set_vars( array(
+		$oTmpl->set_vars( [
 			"title"  => $this->mTitle,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "title" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)